tools/generate_leaderboard_plot.py

```diff
@@ -0,0 +1,117 @@
+# Run this script with `python tools/generate_leaderboard_plot.py`
+# to create the image of the leaderboard heatmap displayed in
+# docs/leaderboards.md
+#
+# Note: the code below assumes that the eva results are stored in
+# `eva/logs/<task>/<fm_identifier>`.
+
+import os
+import json
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.colors
+import seaborn as sns
+
+
+def main():
+    _tasks_to_metric = {
+        "bach": "MulticlassAccuracy",
+        "crc": "MulticlassAccuracy",
+        "mhist": "BinaryBalancedAccuracy",
+        "patch_camelyon": "BinaryBalancedAccuracy",
+        "camelyon16": "BinaryBalancedAccuracy",
+        "panda": "MulticlassAccuracy",
+        "consep": "GeneralizedDiceScore",
+        "monusac": "GeneralizedDiceScore",
+    }
+    _fm_name_map = {
+        "dino_vits16_lunit": "Lunit - ViT-S16 | TCGA",
+        "owkin_phikon": "Owkin (Phikon) - iBOT ViT-B16 | TCGA",
+        "dino_vitl16_uni": "UNI - DINOv2 ViT-L16 | Mass-100k",
+        "dino_vits16_kaiko": "kaiko.ai - DINO ViT-S16 | TCGA",
+        "dino_vits8_kaiko": "kaiko.ai - DINO ViT-S8 | TCGA",
+        "dino_vitb16_kaiko": "kaiko.ai - DINO ViT-B16 | TCGA",
+        "dino_vitb8_kaiko": "kaiko.ai - DINO ViT-B8 | TCGA",
+        "dino_vitl14_kaiko": "kaiko.ai - DINOv2 ViT-L14 | TCGA",
+    }
+    _tasks_names_map = {
+        "bach": "BACH",
+        "crc": "CRC",
+        "mhist": "MHIST",
+        "patch_camelyon": "PCam",
+        "camelyon16": "Camelyon16",
+        "panda": "PANDA",
+        "consep": "CoNSeP",
+        "monusac": "MoNuSAC",
+    }
+
+    # get log_dit from arg parser
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--logs_dir", type=str, default="logs")
+    parser.add_argument("--output_file", type=str, default="docs/images/leaderboard_new.svg")
+    args = parser.parse_args()
+
+    # load existing leaderboard if available:
+    if os.path.isfile("tools/data/leaderboard.csv"):
+        df_existing = pd.read_csv("tools/leaderboard.csv")
+        df_existing = df_existing.set_index("Unnamed: 0", drop=True)
```

Comment:
Should we give this column a name? e.g. `model`, then we can replace the `Unnamed: 0` here

Reply:
done